### ISSUE TITLE
On hx-post, even when the server retrieves a HX-trigger event the client does not react

### ISSUE THREAD

**Issue by vitordeoliveira** [bigskysoftware/htmx#2334]
On the following code, the idea was after the request, the server dispatched an event to refresh the list of todos, but didn't work.... it works file with hx-delete event, but with hx-post in the form nothing happens, is that a bug? or is intentional?

```
            <form class="flex mt-4" hx-post="/todolist">
                <input class="shadow appearance-none border rounded w-full py-2 px-3 mr-4 text-grey-600" id="todo" name="todo" type="text" placeholder="Add Todo">
                <button class="flex-no-shrink p-2 border-2 rounded text-teal-600 border-teal-600 hover:text-white hover:bg-teal-600" id="button" type="submit" >Add</button>
            </form>
```

**Comment by H4rryK4ne** [comment]
I am currently struggeling with the same problem.

I have form
```html
<div id="detail">
<form hx-post="/todolist" hx-target="#detail" hx-swap="innerHTML">
    <input id="todo" name="todo" type="text" placeholder="Add Todo">
    <button id="button" type="submit" >Add</button>
</form>
</div>
``` 

and  another element 
```html
<div id="list" hx-get="/whatever" hx-trigger="my-trigger from:body">...</div>
```

The post request responses with the header `HX-Trigger: {"my-trigger": {}"}`. But it seems not to trigger the event.

Right after the POST request the form will be replaced by a redirect to the created object detail. However the response of this GET request does trigger an event.

**Comment by Telroshan** [comment]
> Right after the POST request the form will be replaced by a redirect to the created object detail.

Is the POST request being redirected? If so, I think the headers you set won't be even caught by htmx at all as the request will natively follow the redirection.
So if I understand correctly that you're making the POST request -> setting the HX-Trigger header -> redirecting the request to another endpoint, then htmx won't even see this HX-Trigger header and only see this redirected endpoint's response headers. In which case, you'll want to set the trigger header from that second endpoint instead of the first.

If not, could you maybe provide more details, i.e. the snippet of your server code that sets the header, or a screenshot of the response in your browser's network tab to confirm the header is indeed in there?
Let me know!

**Comment by H4rryK4ne** [comment]
> Is the POST request being redirected? If so, I think the headers you set won't be even caught by htmx at all as the request will natively follow the redirection.

Hmm ... that would be sad ...

>  you'll want to set the trigger header from that second endpoint instead of the first.
I could do this, but then the event is always triggered when I display the data (e.g. I click on an item in the list, which I want to update with the event) and not only when I create or update the data with the form/POST ... 

To prevent this, I would need the information, that this is a redirect from update or create ... but currently I don't have an idea how to do this.

For me this issue is clarified ... thanks for the quick response!

**Comment by matthewhegarty** [comment]
I have the same issue... adding this code as per the [docs][1]:

```javascript
    document.body.addEventListener("refreshTable", function (evt) {
      console.log("refreshTable was triggered!");
    });
```

The console output is seen in response to the GET but not the POST.  For both GET and POST the status is 200.  The header is definitely in the response in both cases (verified in Chrome dev tools - though it is named 'Hx-Trigger')

[1]: https://htmx.org/headers/hx-trigger/

**Comment by 1cg** [comment]
Can you run an uncompressed version and set a break point here:

https://github.com/bigskysoftware/htmx/blob/3dc73f6d04c856ad68400ee49d43fe87f9e4f164/src/htmx.js#L4630

I don't see why a post vs a get should make any difference in the response processing.

**Comment by Telroshan** [comment]
> My POST is an upload of file data, sent using javascript

If I understand correctly that you're sending the request yourself, i.e. not using htmx attributes nor `htmx.ajax` to do so, then indeed htmx won't know at all about your request nor its response and won't process it at all.
If that's the case, is there any reason not to use [`htmx.ajax`](https://htmx.org/api/#ajax) here?

Hope this helps!

**Comment by matthewhegarty** [comment]
> htmx won't know at all about your request nor its response and won't process it at all.

Yes this is correct, I had assumed that setting the header in the response would mean that htmx logic could respond in the page, but like you say it has to be issued from htmx.  I am trying now to use `htmx.ajax()` but this gives me a `TargetError` which I am trying to figure out.

**Comment by matthewhegarty** [comment]
> I don't see why a post vs a get should make any difference in the response processing.

It's as stated by @Telroshan - because I issue the post from js not htmx.

thanks
